keps/sig-node/2837-pod-level-resource-spec/README.md

```diff
@@ -0,0 +1,1129 @@
+    - name: container-1
+      resources:
+        requests:
+          memory: 100Mi
+          cpu: 1
+    - name: container-2
+  ...
+  ```
+
+  No changes required to support this.
+
+* **Scenario 2: Only pod-level resources are specified**
+
+  ```yaml
+  apiVersion: v1
+  kind: Pod
+  spec:
+    resources:
+      requests:
+        memory: 100Mi
+        cpu: 1
+  ...
+    containers:
+    - name: container-1
+    ...
+    - name: container-2
+  ...
+  ```
+
+  Following validation logic needs to be added for `resources` specified in the PodSpec:
+
+  1. Only CPU and memory resources are supported in pod level resource specifications, hence invalidate pod-level resource specification for other resources 
+  like gpu, storage, etc.
+
+* **Scenario 3: Both pod-level and container level resources are specified.**
+
+  ```yaml
+  apiVersion: v1
+  kind: Pod
+  ...
+  spec:
+  resources:
+    requests:
+      memory: 100Mi
+      cpu: 1
+    containers:
+    - name: container-1
+        resources:
+          requests:
+            memory: 100Mi
+            cpu: 1
+    ...
+    - name: container-2
+  ...
+  ```
+
+  Following validation logic needs to be added:
+
+  1. requests
+      * `pod.spec.resources.requests[memory/cpu] >= sum(pod.spec.containers[*].resources.requests[memory/cpu])` is valid.
+
+        Pod-level resource requests are allowed to be greater than the sum of container level resource requests to account for ephemeral containers that could 
+        be added later.
+
+      * `pod.spec.resources.requests[memory/cpu] < sum(pod.spec.containers[*].resources.requests[memory/cpu])` is invalid.
+
+          This must result in an error as it is semantically inconsistent. Pod-level requests are meant to represent the minimum amount of resources needed for 
+          all the containers within a pod to function correctly. If a pod’s request is lower than the sum of its container requests, it creates a logical 
+          contradiction: the pod as a whole requires less resources than the sum of its parts.
+
+          Besides, the scheduler will use pod requests to determine suitable nodes for placement. If a pod’s request underestimates its actual resource needs
+          (represented by the sum of container requests), it could be scheduled on a node that cannot provide enough resources, leading to failures and 
+          performance degradation.
+
+  2. limits
+      * `pod.spec.resources.limits[memory/cpu] >= sum(pod.spec.containers[*].resources.limits[memory/cpu])` is valid.
+
+          Pod-level resource limits are allowed to be greater than the sum of container level resource limits to account for ephemeral containers that could be added 
+          later. Setting a resource limit on the pod in its cgroup settings will not reserve any resource and hence not affect the other workloads running on the node.
+
+      * `pod.spec.resources.limits[memory/cpu] < sum(pod.spec.containers[*].resources.limits[memory/cpu])` is valid.
```

Comment:
That should already be true in all cases

Reply:
I don't think so, since pod limits are allowed to be smaller than the sum of container limits. If you set a pod limit, and omit pod requests, I think you could end up with this:

pod limit: 10
pod requests: nil
container 1 limit: 10
container 1 request: 10
container 2 limit: 10
container 2 request: 10

If we required pod requests to be set whenever pod limits are set, then I think it holds.

Reply:
Something like `pod.spec.resources.limits[memory/cpu] >= minimum_effective_pod_requests[memory/cpu]`

Reply:
> When request is not specified, it defaults to the limit, right?

yes, that's true for containers and we should apply the same logic to pods. @ndixita